### PR TITLE
Update Windows cross-compile dependencies

### LINKED ISSUE #5718: Move travis nightlies to toxcore 0.2.9
@sudden6 
travis nightlies use toxcore 0.2.8. I decided to try one of the "official" travis nightlies, and all that happened is my persistent groups were removed because of the old version of toxcore.

screen3

I'm still using the Jenkins builds right now for nightlies. Also, while you update toxcore to the latest version, I'll go ahead and ask both you and @nurupo to update Qt to 5.9.8.

Thanks!

### PR DESCRIPTION
**Pull request by nurupo**
Fixes #5718

Tested the changes locally for both release x86_64 and i686 - they build, qTox starts up, I can get to About page and see updated version numbers. Haven't tested anything else, so no idea if it breaks anything.

Feel free to squash commits if you'd like. I made them separate so that's it's easier to revert a specific dependency update if needed be.

---
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/5725)

### CODE REVIEW

**Comment by nurupo** [comment]
Force-pushed to add a missing word in a commit message.

File: windows/cross-compile/build.sh [hunk 1]
```diff
@@ -388,15 +389,15 @@ set -u
 # SQLCipher
 
 SQLCIPHER_PREFIX_DIR="$DEP_DIR/libsqlcipher"
-SQLCIPHER_VERSION=v3.4.2
-SQLCIPHER_HASH="69897a5167f34e8a84c7069f1b283aba88cdfa8ec183165c4a5da2c816cfaadb"
+SQLCIPHER_VERSION=v4.2.0
```
Comment (sudden6) [inline windows/cross-compile/build.sh:392]: @anthonybilinski does switching to sqlcipher 4.x break compatibility also on windows? IIRC you did some stuff in that direction, but I can't remember exactly what.

**Comment by nurupo** [comment]
Rebased on master.

Turns out @anthonybilinski has bumped Toxcore to 0.2.9 right after I started working on this branch, so rebasing removed the redundant https://github.com/qTox/qTox/pull/5725/commits/dbbe8e83459999ef17be41911ae855c7f144fb02 I have made.

Toxcore just released 0.2.10, so I set the build script to use that instead.

**Comment by nurupo** [comment]
`ci/circleci: deps64` hit a bad Qt mirror:
```
+ wget --timeout=10 https://download.qt.io/official_releases/qt/5.9/5.9.8/single/qt-everywhere-opensource-src-5.9.8.tar.xz
--2019-06-25 18:56:56--  https://download.qt.io/official_releases/qt/5.9/5.9.8/single/qt-everywhere-opensource-src-5.9.8.tar.xz
Resolving download.qt.io (download.qt.io)... 77.86.229.90
Connecting to download.qt.io (download.qt.io)|77.86.229.90|:443... connected.
Unable to establish SSL connection.
Exited with code 4
```

Can someone restart it?

**Comment by sudden6** [comment]
Restarted CircleCI

**Review by anthonybilinski — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5725)* status: 0 of 1 LGTMs obtained

___
*[windows/cross-compile/build.sh, line 392 at r1](https://reviewable.io/reviews/qtox/qtox/5725#-LiC4FylKVftySQFaRye-r1-392:-LiFxiBw1wE2UiCnxJHf:b-e77hww) ([raw file](https://github.com/qtox/qtox/blob/92b740cbbb0fe9391266e13315a4d941e5a11aea/windows/cross-compile/build.sh#L392)):*
<details><summary><i>Previously, sudden6 wrote…</i></summary><blockquote>

@anthonybilinski does switching to sqlcipher 4.x break compatibility also on windows? IIRC you did some stuff in that direction, but I can't remember exactly what.
</blockquote></details>

We used to use sqlcipher defaults, which changed from 3.x to 4.x and broke our decryption. I changed RawDatabase to explicitly use 3.x defaults (regardless of if we're running 4.x), so this upgrade is safe.

**Comment by nurupo** [comment]
Fun fact: toxcore's toxencryptsave [has the same issue](https://github.com/TokTok/c-toxcore/issues/653). It's using libsodium's default options for scrypt which are subject to change in newer versions of sodium, so it will break sometime in the future.

**Review by sudden6 — APPROVED** [review]
:lgtm_strong: then

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/5725#-:-LiHUTmk0rMoxT0KOgze:b1hmrgw)* status: :shipit: complete! 1 of 1 LGTMs obtained

**Pull request merged**